game-core/src/test/java/games/strategy/engine/data/RouteFinderTest.java

```diff
@@ -0,0 +1,72 @@
+package games.strategy.engine.data;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RouteFinderTest {
+
+  private final PlayerID player = mock(PlayerID.class);
+  private final GameMap map = mock(GameMap.class);
+  private final List<Territory> territories = new ArrayList<>();
+
+  private final int[][] graph = {
```

Comment:
Would an ASCII-art rendition of the graph you drew in the description be helpful for a future code reader to understand this matrix?

Reply:
Yes, however I wasn't able to find a tool that auto-generate one for me so far ^^
So if you can give me a finished ascii art, I'd put it in there as a comment